pandas/core/frame.py

```diff
@@ -515,24 +516,38 @@ def __init__(
             else:
                 mgr = init_dict({}, index, columns, dtype=dtype)
         else:
-            try:
-                arr = np.array(data, dtype=dtype, copy=copy)
-            except (ValueError, TypeError) as err:
-                exc = TypeError(
-                    "DataFrame constructor called with "
-                    f"incompatible data and dtype: {err}"
-                )
-                raise exc from err
+            if not dtype:
+                dtype, _ = infer_dtype_from_scalar(data, pandas_dtype=True)
 
-            if arr.ndim == 0 and index is not None and columns is not None:
-                values = cast_scalar_to_array(
-                    (len(index), len(columns)), data, dtype=dtype
-                )
-                mgr = init_ndarray(
-                    values, index, columns, dtype=values.dtype, copy=False
-                )
+            if is_extension_array_dtype(dtype):
+                if index is None or columns is None:
+                    raise ValueError("DataFrame constructor not properly called!")
+
+                values = [
+                    construct_1d_arraylike_from_scalar(data, len(index), dtype)
+                    for _ in range(len(columns))
+                ]
+                mgr = arrays_to_mgr(values, columns, index, columns, dtype=None)
             else:
-                raise ValueError("DataFrame constructor not properly called!")
+                try:
```

Comment:
can you comment on what is happening here

also this is basically a straight move?

Reply:
Yeah, this already existed here, not totally sure what the intent is. If I had to guess, I'd say it's creating an array and then a few lines down checks to see if the arrays dimensionality is 0, in effect confirming that data is a scalar

Reply:
if this is a straight move that is ok. just a general comment, like 'attempt to coerce to a numpy array'

Reply:
@jreback I see this response, but do you have anything else that you want to see in this PR? Would like to group some other changes together, if necessary, so I don't have to create a commit just to add a single comment haha

Reply:
Added this comment